Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs

```diff
@@ -42,32 +40,46 @@ import Tools.Error
 import qualified Tools.Notifications as Notify
 
 data DUpdateReq
-  = PaymentCompletedReq
-      { bookingId :: Id DBooking.Booking,
-        rideId :: Id DRide.Ride,
-        paymentStatus :: PaymentStatus,
-        paymentMethodInfo :: DMPM.PaymentMethodInfo
-      }
-  | EditLocationReq
-      { bookingId :: Id DBooking.Booking,
-        rideId :: Id DRide.Ride,
-        origin :: Maybe Common.Location,
-        destination :: Maybe Common.Location
-      }
-  | AddStopReq
-      { bookingId :: Id DBooking.Booking,
-        stops :: [Common.Location]
-      }
-  | EditStopReq
-      { bookingId :: Id DBooking.Booking,
-        stops :: [Common.Location]
-      }
+  = UPaymentCompletedReq PaymentCompletedReq
+  | UEditLocationReq EditLocationReq
+  | UAddStopReq AddStopReq
+  | UEditStopReq EditStopReq
+
+data PaymentCompletedReq = PaymentCompletedReq
+  { bookingId :: Id DBooking.Booking,
+    rideId :: Id DRide.Ride,
+    paymentStatus :: PaymentStatus,
+    paymentMethodInfo :: DMPM.PaymentMethodInfo
+  }
+
+data EditLocationReq = EditLocationReq
+  { bookingId :: Id DBooking.Booking,
+    rideId :: Id DRide.Ride,
+    origin :: Maybe Common.Location,
+    destination :: Maybe Common.Location
+  }
+
+data AddStopReq = AddStopReq
+  { bookingId :: Id DBooking.Booking,
+    stops :: [Common.Location]
+  }
+
+data EditStopReq = EditStopReq
+  { bookingId :: Id DBooking.Booking,
+    stops :: [Common.Location]
+  }
+
+getBookingId :: DUpdateReq -> Id DBooking.Booking
+getBookingId (UPaymentCompletedReq req) = req.bookingId
+getBookingId (UEditLocationReq req) = req.bookingId
+getBookingId (UAddStopReq req) = req.bookingId
+getBookingId (UEditStopReq req) = req.bookingId
 
 data PaymentStatus = PAID | NOT_PAID
 
 handler :: DUpdateReq -> Flow ()
-handler req@PaymentCompletedReq {} = do
-  unless (req.paymentMethodInfo.paymentType == DMPM.ON_FULFILLMENT) $
+handler (UPaymentCompletedReq req@PaymentCompletedReq {}) = do
+  unless (req.paymentMethodInfo.paymentType == DMPM.POSTPAID) $
```

Comment:
It needs to be `ON_FULFILLMENT`. Lets remove POSTPAID, it has been removed from production.